server_monitoring/models/server_monitor_process.py

```diff
@@ -0,0 +1,248 @@
+
+import psutil
+
+import openerp
+from openerp.http import request
+from openerp import models, fields, api
+from openerp import pooler
+from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+_logger = logging.getLogger(__name__)
+
+BLACKLIST = (
+    type, tuple, dict, list, set, frozenset,
+    property,
+    classmethod,
+    staticmethod,
+    types.FunctionType,
+    types.ClassType,
+    types.ModuleType, types.FunctionType, types.MethodType,
+    types.MemberDescriptorType, types.GetSetDescriptorType,
+    )
+
+_monkey_patched = False
+
+
+class ClassInstanceCount(models.Model):
+    _name = 'server.monitor.class.instance.count'
+
+    name = fields.Char('Class name', readonly=True)
+    count = fields.Bigint('Instance count', readonly=True)
+    measure_id = fields.Many2one('server.monitor.process',
+                                 'Measure',
+                                 readonly=True,
+                                 ondelete='cascade')
+
+
+def _monkey_patch_object_proxy_execute():
+    orig_execute_cr = openerp.service.model.execute_cr
+
+    def execute_cr(cr, uid, obj, method, *args, **kw):
+        result = orig_execute_cr(cr, uid, obj, method, *args, **kw)
+        pool = pooler.get_pool(cr.dbname)
+        monitor_obj = pool.get('server.monitor.process')
+        monitor = pool['ir.config_parameter'].get_param(
+            cr, uid,
+            'server_monitoring.monitor_rpc_calls', default=False
+        )
+        if monitor_obj is not None and bool(monitor):
+            monitor_obj.log_measure(cr, uid, obj, method, 'xmlrpc call',
+                                    False, False, context={})
+        return result
+
+    openerp.service.model.execute_cr = execute_cr
+
+
+def _monkey_patch_controller_call_kw():
+    orig_call_kw = openerp.addons.web.controllers.main.DataSet._call_kw
+
+    def _call_kw(self, model, method, args, kwargs):
+        result = orig_call_kw(self, model, method, args, kwargs)
+        monitor_obj = request.registry.get('server.monitor.process')
+        monitor = request.registry['ir.config_parameter'].get_param(
+            request.cr, request.uid,
+            'server_monitoring.monitor_rpc_calls', default=False
+        )
+        if monitor_obj is not None and bool(monitor):
+            monitor_obj.log_measure(request.cr, request.uid, model, method,
+                                    'jsonrpc call',
+                                    False, False, context={})
+        return result
+
+    openerp.addons.web.controllers.main.DataSet._call_kw = _call_kw
+
+
+class ServerMonitorProcess(models.Model):
+    def __init__(self, pool, cr):
+        super(ServerMonitorProcess, self).__init__(pool, cr)
+
+    _name = 'server.monitor.process'
+
+    def _register_hook(self, cr):
+        global _monkey_patched
+        if _monkey_patched:
+            return
+        _monkey_patched = True
+        _monkey_patch_controller_call_kw()
+        _monkey_patch_object_proxy_execute()
```

Comment:
You should use [`_patch_method` (and possibly `_revert_method`)](https://github.com/odoo/odoo/blob/8.0/openerp/models.py#L5211-L5247) instead, as they are the standards for monkey patching.

These methods should be found/returned inside the class to allow inheritance.